SIL.Core.Tests/Extensions/StringExtensionTests.cs

```diff
@@ -716,6 +720,79 @@ public void GetLongestUsefulCommonSubstring_SurrogatePairsWithSymbolsAsWordBreak
 				out var _, minPctForPartialWordMatch));
 		}
 
+		[TestCase("\u200D")] // Format: ZWJ
+		[TestCase("s")] // Lowercase letter
+		[TestCase("\u02C6")] // Modifier letter
+		[TestCase("\u02C6")] // NonSpacingMark
+		[TestCase("\u02C6")] // OtherLetter
+		[TestCase("\u100000")] // PrivateUse
+		[TestCase("\u094C")] // SpacingCombiningMa
+		[TestCase("\u01C8")] // TitlecaseLetter:
+		[TestCase("Q")] // Uppercase letter
+		public void IsLikelyWordForming_WordformingCharactersKnownToDotNet_ReturnsTrue(string s)
+		{
+			Assert.That(StringExtensions.AltImplGetUnicodeCategory, Is.Null);
+			Assert.IsTrue(s.IsLikelyWordForming(0, false));
+		}
+
+
+		[TestCase(")")] // ClosePunctuation
+		[TestCase("\u203F")] // ConnectorPunctuation
+		[TestCase("\u0000")] // Control
+		[TestCase("$")] // CurrencySymbol
+		[TestCase("-")] // DashPunctuation
+		[TestCase("5")] // DecimalDigitNumber
+		[TestCase("\u20DD")] // EnclosingMark
+		[TestCase("»")] // FinalQuotePunctuation
+		[TestCase("\u201C")] // InitialQuotePunctuation
+		[TestCase("\u2129")] // LetterNumber
+		[TestCase("\u2028")] // LineSeparator
+		[TestCase("+")] // MathSymbol
+		[TestCase("\u0060")] // ModifierSymbol
+		[TestCase("[")] // OpenPunctuation
+		[TestCase("\u00B2")] // OtherNumber
+		[TestCase("!")] // OtherPunctuation
+		[TestCase("\u00B0")] // OtherSymbol
+		[TestCase("\u2029")] // ParagraphSeparator
+		[TestCase(" ")] // SpaceSeparator
+		public void IsLikelyWordForming_OtherCharactersKnownToDotNet_ReturnsFalse(string s)
+		{
+			Assert.That(StringExtensions.AltImplGetUnicodeCategory, Is.Null);
+			Assert.IsFalse(s.IsLikelyWordForming(0, false));
+		}
+
+		[TestCase("\u00016FF0")] // VIETNAMESE ALTERNATE READING MARK CA
+		public void IsLikelyWordForming_CharactersNotKnownToDotNet_ReturnsFalse(string s)
+		{
+			Assert.That(StringExtensions.AltImplGetUnicodeCategory, Is.Null);
+			var result = s.IsLikelyWordForming(0, false);
+			if (result)
+			{
+				Assert.Ignore("This test was written to demonstrate the need for an alternate " +
+					"implementation for characters that are not known to .Net (see next test), " +
+					"but it is possible that some future .Net version will come out that knows " +
+					"about this code point.");
+			}
+		}
+
+		[Test]
+		public void IsLikelyWordForming_CharactersNotKnownToDotNetButHandledByAlternateImplementation_ReturnsTrue()
+		{
+			const string vietnameseAlternateReadingMark = "\u00016FF0";
+			StringExtensions.AltImplGetUnicodeCategory = (s, i) =>
+				s == vietnameseAlternateReadingMark ? UnicodeCategory.SpacingCombiningMark :
+				CharUnicodeInfo.GetUnicodeCategory(s, i);
+
+			try
+			{
+				Assert.IsTrue(vietnameseAlternateReadingMark.IsLikelyWordForming(0, false));
+			}
+			finally
+			{
+				StringExtensions.AltImplGetUnicodeCategory = null;
```

Comment:
We could also do this in a setup method that runs before each test. That way we wouldn't have to check for `StringExtensions.AltImplGetUnicodeCategory` being `null` at the start of each test, and we wouldn't need this try/finally block here.